tests/unit/formats/test_cvat.py

```diff
@@ -0,0 +1,199 @@
+                </task>
+              </meta>
+              <image id="0" name="img0.jpg" width="10" height="8">
+                <box label="label1" xtl="invalid" ytl="0.0" xbr="5.0" ybr="2.0"></box>
+              </image>
+          </annotations>
+        """
+        xml_path = create_xml_file(tmp_path, invalid_annotation)
+
+        with pytest.raises(
+            ValueError,
+            match="Input should be a valid number, unable to parse string as a number",
+        ):
+            label_input = CVATObjectDetectionInput(xml_path)
+            list(label_input.get_labels())
+
+    def test_missing_attributes_for_image(self, tmp_path: Path) -> None:
+        invalid_annotation = """
+          <annotations>
+              <meta>
+                <task>
+                  <labels>
+                    <label><name>label1</name></label>
+                  </labels>
+                </task>
+              </meta>
+              <image id="0" name="img0.jpg" width="10" >
+                <box label="label1" xtl="0.0" ytl="0.0" xbr="5.0" ybr="2.0"></box>
+              </image>
+          </annotations>
+        """
+        xml_path = create_xml_file(tmp_path, invalid_annotation)
+
+        with pytest.raises(
+            ValueError,
+            match="validation error for CVATAnnotations\nimages.0.height",
+        ):
+            label_input = CVATObjectDetectionInput(xml_path)
+            list(label_input.get_labels())
+
+    def test_invalid_label(self, tmp_path: Path) -> None:
+        invalid_annotation = """
+          <annotations>
+              <meta>
+                <task>
+                  <labels>
+                    <label><name>label1</name></label>
+                  </labels>
+                </task>
+              </meta>
+              <image id="0" name="img0.jpg" width="10" height="8">
+                <box label="label2" xtl="1.0" ytl="0.0" xbr="5.0" ybr="2.0"></box>
+              </image>
+          </annotations>
+        """
+        xml_path = create_xml_file(tmp_path, invalid_annotation)
+
+        with pytest.raises(ParseError, match="Unknown category name 'label2'"):
+            label_input = CVATObjectDetectionInput(xml_path)
+            list(label_input.get_labels())
+
+
+def _compare_xml_elements(elem1: ET.Element, elem2: ET.Element) -> bool:
+    """Recursively compare two XML elements for tag, attributes, and text."""
+    if elem1.tag != elem2.tag or elem1.text != elem2.text:
+        return False
+
+    if elem1.attrib != elem2.attrib:
+        return False
+
+    children1 = list(elem1)
+    children2 = list(elem2)
+
+    if len(children1) != len(children2):
+        return False
+
+    return all(_compare_xml_elements(c1, c2) for c1, c2 in zip(children1, children2))
+
+
+class TestCVATObjectDetectionOutput:
+    @pytest.mark.parametrize("annotation_scope", ["task", "project", "job"])
+    def test_cyclic_load_save(self, tmp_path: Path, annotation_scope: str) -> None:
```

Comment:
The naming of the test is not correct, see https://www.notion.so/getlightly/Python-testing-78e2dc49baf340178fa44ce20031c52e?pvs=4#2066d6ffc3a04bd7b431ab280b1e611c
```suggestion
    def test_save_cyclic_load_and_save(self, tmp_path: Path, annotation_scope: str) -> None:
```